gnxi_tester/config/history_test.go

```diff
@@ -30,6 +30,158 @@ type Want struct {
 	err     error
 }
 
+func TestSetTarget(t *testing.T) {
+	viper.SetConfigFile("/tmp/config.yml")
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Errorf("Error getting working directory: %v", err)
+	}
+	certPath := path.Join(dir, "ca.crt")
+	certKeyPath := path.Join(dir, "ca.key")
+	tests := []struct {
+		name          string
+		targetName    string
+		targetAddress string
+		targetCA      string
+		targetCAKey   string
+		config        map[string]Target
+		lastTarget    string
+		want          Want
+	}{
+		{
+			name: "No targets in history, no target specified",
+			want: Want{
+				err:     errors.New("No targets in history and no target specified"),
+				targets: map[string]Target{},
+			},
+		},
+		{
+			name: "No target specified",
+			config: map[string]Target{"myhost.com": {
+				Address: "localhost:9339",
+				Ca:      certPath,
+				CaKey:   certKeyPath,
+			}},
+			lastTarget: "myhost.com",
+			want: Want{
+				err: nil,
+				targets: map[string]Target{"myhost.com": {
+					Address: "localhost:9339",
+					Ca:      certPath,
+					CaKey:   certKeyPath,
+				}},
+			},
+		},
+		{
+			name: "Non-existent target",
+			config: map[string]Target{"myhost.com": {
+				Address: "localhost:9339",
+				Ca:      certPath,
+				CaKey:   certKeyPath,
+			}},
+			targetName: "nonexistenttarget",
+			lastTarget: "myhost.com",
+			want: Want{
```

Comment:
since you're using this variable for both want and got I would probably say renaming would not be a terrible idea. First I did was trying to understand whether got:= Want{... was a mistake and I feel other may end up spending time doing the same which ... is not ideal! May I suggest "result" or something like that? Also have a comment describing what is it near definition of exported structs please.

Reply:
Agreed! Changing, thank you!